### PR TITLE
Add Plugin Customizable AV1 Transcode

### PR DESCRIPTION
**Pull request by Jessomadic**

### CODE REVIEW

**Review by Jessomadic — COMMENTED** [review]
Looks Good

**Comment by Jessomadic** [comment]
Ok now I'm done

**Comment by Jessomadic** [comment]
@HaveAGitGat

**Comment by Jessomadic** [comment]
Don't merge discovered a pretty big issue that I will look at tomorrow

**Comment by Jessomadic** [comment]
ok that fixed it

**Comment by Jessomadic** [comment]
Photos for examples that muti files can be transcoded at once while changing the settings if needed 
image
image
image

**Comment by Jessomadic** [comment]
Sorry for all the commits I just want it to be perfect

**Review by Jessomadic — COMMENTED** [review]
Reviewed Changes. Been running plugin for 2 days with no issues

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
+/* eslint no-plusplus: ["error", { "allowForLoopAfterthoughts": true }] */
+/* eslint-disable no-restricted-globals */
+const details = () => ({
+  id: 'Tdarr_Plugin_AV1_HandBreak_Transcode',
+  Stage: 'Pre-processing',
+  Name: 'AV1 HandBreak Transcoder',
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:6]: Handbreak should be handbrake

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  Type: 'Video',
+  Operation: 'Transcode',
+  Description: 'Transcodes to AV1 at the selected Bitrate. This is best used with Remux Files.',
+  Version: '2.1.2',
+  Tags: 'HandBrake,AV1',
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:11]: Tag options: `h265,hevc,h264,nvenc h265,nvenc h264,video only,audio only,subtitle only,handbrake,ffmpeg,radarr,sonarr,pre-processing,post-processing,configurable`

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  else {
+    // loop through the resolution order
+    for (let i = 0; i < resolutionOrder.length; i += 1) {
+      // if the selected resolution is the same as the current resolution in the loop
+      if (selectedResolution === resolutionOrder[i]) {
+        // break the loop
+        break;
+      }
+    }
+  }
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:147]: Entire else does nothing?

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    inputs.BitRate = videoBitRate;
+    // eslint-disable-next-line brace-style
+  }
+  // if VideoBitrate is larger than selected bitrate then use the selected bitrate
+  else {
+    // eslint-disable-next-line no-self-assign, no-param-reassign
+    inputs.BitRate = inputs.BitRate;
+  }
+
+  response.infoLog += '';
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:168]: Empty infoLog +=

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  // eslint-disable-next-line no-constant-condition
+  if ((true) || file.forceProcessing === true) {
+    // eslint-disable-next-line max-len
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:171]: Nothing stops the loop?

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    response.preset = `--encoder svt_av1 -b ${inputs.BitRate} -r 24 -E aac -B 320 -R Auto -6 dpl2 --normalize-mix 1 -f ${inputs.Container} --no-optimize ${dimensions} --crop 0:0:0:0`;
+    response.container = `.${inputs.Container}`;
+    response.handbrakeMode = true;
+    response.ffmpegMode = false;
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:175]: Don't need to set these

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+  else {
+    // eslint-disable-next-line no-self-assign, no-param-reassign
+    inputs.BitRate = inputs.BitRate;
+  }
+
+  response.infoLog += '';
+  // eslint-disable-next-line no-constant-condition
+  if ((true) || file.forceProcessing === true) {
+    // eslint-disable-next-line max-len
+    response.preset = `--encoder svt_av1 -b ${inputs.BitRate} -r 24 -E aac -B 320 -R Auto -6 dpl2 --normalize-mix 1 -f ${inputs.Container} --no-optimize ${dimensions} --crop 0:0:0:0`;
```
Comment (supersnellehenk) [inline Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:172]: Why is it cropping for 0? Also, plugin doesn't mention it's going to convert everything to AAC or anything else that it's doing in this command.

File: tests/Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
+/* eslint no-plusplus: ["error", { "allowForLoopAfterthoughts": true }] */
```
Comment (supersnellehenk) [inline tests/Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:1]: You don't paste the entire plugin into the tests folder, please look at other tests.

**Review by supersnellehenk — CHANGES_REQUESTED** [review]

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js (hunk 7 above)
Comment (Jessomadic) [reply Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:172]: From what I understand Handbrake tries to auto-crop video and this was the way I could think of to disable that if that is incorrect I will just remove it from the preset

**Comment by Jessomadic** [comment]
Not all requested changes have been met yet

File: Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js (hunk 5 above)
Comment (Jessomadic) [reply Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js]: Oops resolved in next push

File: tests/Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js (hunk 8 above)
Comment (Jessomadic) [reply tests/Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js:1]: I'm looking at other tests but I don't grasp how it should be written or when to use the required files. Sorry, I'm still fairly new to programming.

**Comment by supersnellehenk** [comment]
@Jessomadic you don't have to keep commenting the current state, just comment on the issues if need be. I won't review it until you request it again.

**Pull request merged**